### PR TITLE
[refactoring] improvements for AptosVM 

### PR DESCRIPTION
**Pull request by georgemitenkov**
### Description

1) .0 --> .vm_impl (we should get rid of _impl in the future)
2) removed unused & cloning APIs
3) is_simulation flag is introduced to clean up code. Simulation wrapper will be removed 
     in a separate PR.

### CODE REVIEW

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 1]
```diff
@@ -1856,34 +1829,30 @@ impl VMValidator for AptosVM {
...
-                VMValidatorResult::new(None, txn.gas_unit_price()),
-            ),
-            Err(err) => (
-                "failure",
-                VMValidatorResult::new(Some(err.status_code()), 0),
-            ),
-        };
+        let (counter_label, result) =
+            match self.validate_signed_transaction(&mut session, &resolver, &txn, &log_context) {
+                Err(err) if err.status_code() != StatusCode::SEQUENCE_NUMBER_TOO_NEW => (
```
Comment (runtian-zhou) [inline aptos-move/aptos-vm/src/aptos_vm.rs:1847]: Thank you!!

**Review by runtian-zhou — APPROVED** [review]

**Review by vgao1996 — APPROVED** [review]
The changes look good, but let's still run replay before checking this in. @georgemitenkov do you know how to trigger a replay task?

**Comment by georgemitenkov** [comment]
@vgao1996 mainnet replay passed, testnet had a couple of time outs (I guess some CI issue, does not seem related to this PR). Should be good to land?

**Comment by vgao1996** [comment]
@georgemitenkov yeah it's known to me that replay has been quite unstable these days. I've also been running into time outs multiple times. Great to hear that everything passed in the 2nd run

**Pull request merged**